File: Hashmap/Word Pattern/WordPattern.cs
public class Solution {
    public bool WordPattern(string pattern, string s) {
        Dictionary<char, int> patternMap = new Dictionary<char, int>();
        Dictionary<string, int> wordMap = new Dictionary<string, int>();
        string[] words = s.Split(' ');
        int i = 0, n = pattern.Length;

        if (n != words.Length) {
            return false;
        }

        for (i = 0; i < n; i++) {
            char currPatternChar = pattern[i];
            string currWord = words[i];
            if (!patternMap.ContainsKey(currPatternChar)) {
                patternMap[currPatternChar] = i;
            }
            if (!wordMap.ContainsKey(currWord)) {
                wordMap[currWord] = i;
            }
            if (patternMap[currPatternChar] != wordMap[currWord]) {
                return false;
            }
        }
        return true;
    }
}

File: Hashmap/Word Pattern/WordPattern.go
func wordPattern(pattern string, s string) bool {
	patternMap := make(map[rune]int)
	wordMap := make(map[string]int)
	words := strings.Split(s, " ")
	n := len(pattern)

	if n != len(words) {
		return false
	}

	for i := 0; i < n; i++ {
		currPatternChar := rune(pattern[i])
		currWord := words[i]

		if _, exists := patternMap[currPatternChar]; !exists {
			patternMap[currPatternChar] = i
		}
		if _, exists := wordMap[currWord]; !exists {
			wordMap[currWord] = i
		}
		if patternMap[currPatternChar] != wordMap[currWord] {
			return false
		}
	}
	return true
}